src/ble/CHIPBleServiceData.h

```diff
@@ -64,17 +66,31 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint16_t GetAdvertisementVersion() const
+    {
+        return chip::Encoding::LittleEndian::Get16(DeviceDiscriminatorAndAdvVersion) & kAdvertisementMask;
+    }
+
+    // Use only 4 bits to set advertisement version
+    void SetAdvertisementVersion(uint8_t advertisementVersion)
+    {
+        // Advertisement Version is 4 bit long from 12th to 15th
+        advertisementVersion &= kAdvertisementVersionMask;
+        advertisementVersion <<= 4u;
+        DeviceDiscriminatorAndAdvVersion[1] |= advertisementVersion;
```

Comment:
So if I `SetAdvertisementVersion(1)` and then `SetAdvertisementVersion(2)`... what value will I get from `GetAdvertisementVersion()` and why?

Reply:
Now it will return actual advertisement version 1, 2 etc.